web-ui/src/main/resources/catalog/components/viewer/wmsimport/WmsImportDirective.js

```diff
@@ -493,6 +510,79 @@
         }
       };
     }]);
+
+
+  /**
+   * @ngdoc directive
+   * @name gn_wmsimport.directive:gnEsriRestCapTreeCol
+   *
+   * @description
+   * Directive to manage a collection of nested layers from
+   * the ESRI REST capabilities document. This directive works with
+   * gnEsriCapTreeElt directive.
+   */
+  module.directive('gnEsriRestCapTreeCol', [
+    '$translate',
+    function($translate) {
+
+      var label= $translate.instant('filter');
+
+      return {
+        restrict: 'E',
+        replace: true,
+        scope: {
+          collection: '='
+        },
+        template: "<ul class='gn-layer-tree'><li data-ng-show='collection.length > 10' >" +
+          "<div class='input-group input-group-sm'><span class='input-group-addon'><i class='fa fa-filter'></i></span>" +
+          "<input class='form-control' aria-label='" + label + "' data-ng-model-options='{debounce: 200}' data-ng-model='layerSearchText'/></div>" +
+          "</li>" +
+          '<gn-esri-rest-cap-tree-elt ng-repeat="member in collection | filter:layerSearchText | orderBy: \'name\'" member="member">' +
+          '</gn-esri-rest-cap-tree-elt></ul>'
+      };
+    }]);
+
+  /**
+   * @ngdoc directive
+   * @name gn_wmsimport.directive:gnEsriCapTreeElt
+   *
+   * @description
+   * Directive to manage layers from an ESRI REST capabilities
+   * document. Will call its own template to display the layer.
+   */
+  module.directive('gnEsriRestCapTreeElt', [
+    '$compile',
+    '$translate',
+    'gnAlertService',
+    function($compile, $translate, gnAlertService) {
+      return {
+        restrict: 'E',
+        require: '^gnWmsImport',
+        replace: true,
+        scope: {
+          member: '='
+        },
+        templateUrl: '../../catalog/components/viewer/wmsimport/' +
+          'partials/esrilayer.html',
+        link: function(scope, element, attrs, controller) {
+          var el = element;
+
+          scope.toggleNode = function(evt) {
+            el.find('.fa').first().toggleClass('fa-folder-open-o')
+              .toggleClass('fa-folder-o');
+            el.children('ul').toggle();
+            evt.stopPropagation();
+          };
+
+          scope.addLayer = function(c) {
+            controller.addEsriRestLayer(scope.member);
+          };
+
+          scope.isParentNode = false;
```

Comment:
This is weird, looks like this directive is built to handle nested layers, but does not actually support it. IIRC layers in ESRI map services are not organized in a tree like in the WMS, so I guess you could get rid of everything related to layer groups and simplify the whole thing.

Future maintainers will thank you.

Reply:
It was copied from other directive for WMS, will cleanup this as I was not sure if nested was supported. I leaved in case that is supported to extend it later.